packages/internal-test-env/index.ts

```diff
@@ -0,0 +1,357 @@
+
+  // VC service provider
+  E2E_TEST_VC_PROVIDER: string;
+
+  // Client credentials for node service
+  E2E_TEST_CLIENT_ID: string;
+  E2E_TEST_CLIENT_SECRET: string;
+
+  // Client credentials for the access requestor
+  E2E_TEST_REQUESTOR_CLIENT_ID: string;
+  E2E_TEST_REQUESTOR_CLIENT_SECRET: string;
+
+  // Client credentials for the resource owner
+  E2E_TEST_RESOURCE_OWNER_CLIENT_ID: string;
+  E2E_TEST_RESOURCE_OWNER_CLIENT_SECRET: string;
+
+  E2E_TEST_FEATURE_FLAG_ACP: boolean | undefined;
+  E2E_TEST_FEATURE_FLAG_WAC: boolean | undefined;
+}
+
+let envLoaded = false;
+let featuredFlags: FeatureFlags = {};
+
+export function setupEnv() {
+  // If we're in CI, the environment is already configured.
+  if (process.env.CI) {
+    return;
+  }
+
+  if (envLoaded) {
+    return;
+  }
+
+  const envPath = join(process.cwd(), "e2e/env");
+
+  // Otherwise load dotenv configuration
+  config({
+    path: envPath,
+    silent: true,
+  });
+
+  if (!process.env.E2E_TEST_ENVIRONMENT) {
+    console.error(
+      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
+tried looking in the following directory for \`.env.local \`: 
+${envPath}`
+    );
+  }
+
+  // Creating feature flag object if there are feature flags
+  Object.keys(process.env)
+    .filter((envVar) => /E2E_TEST_FEATURE.*/.test(envVar))
+    .forEach((key) => (featuredFlags[key] = process.env[key]));
+  // Marking env as loaded
+  envLoaded = true;
+}
+
+function getTestingEnvironment(
+  environment: unknown
+): asserts environment is EnvVariables {
+  // Populate your process.env from your .env file of choice
+
+  // TODO: Replace these inline validations and checks with envalid or env-var
+  if (
+    !availableEnvironment.includes(
+      (environment as EnvVariables).E2E_TEST_ENVIRONMENT as AvailableEnvironment
+    )
+  ) {
+    throw new Error(
+      `Unknown environment: [${
+        (environment as EnvVariables).E2E_TEST_ENVIRONMENT
+      }]`
+    );
+  }
+
+  if (
+    !availableProtocol.includes(
+      (environment as EnvVariables)
+        .E2E_TEST_NOTIFICATION_PROTOCOL as AvailableProtocol
+    )
+  ) {
+    throw new Error(
+      `Unknown protocol: [${
+        (environment as EnvVariables).E2E_TEST_NOTIFICATION_PROTOCOL
+      }]`
+    );
+  }
+
+  if (typeof (environment as EnvVariables).E2E_TEST_IDP !== "string") {
+    throw new Error("The environment variable E2E_TEST_IDP is undefined.");
+  }
+
+  if (
+    typeof (environment as EnvVariables).E2E_TEST_NOTIFICATION_GATEWAY !==
+    "string"
+  ) {
+    throw new Error(
+      "The environment variable E2E_TEST_NOTIFICATION_GATEWAY is undefined."
+    );
+  }
+}
+
+export function getNodeTestingEnvironment(
+  features?: FeatureFlags
+): TestingEnvironmentNode {
+  setupEnv();
+  getTestingEnvironment(process.env);
+
+  if (
+    process.env.E2E_TEST_REQUESTOR_CLIENT_ID !== undefined ||
+    process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== undefined
```

Comment:
As mentioned earlier for the gateway, these env vars are specific to a certain library, the environment won't validate in most of our repos as is.

Reply:
Okay I can remove this  validation then